mutiny-server/src/routes.rs

```diff
@@ -0,0 +1,526 @@
+        .close_channel(&outpoint, address, false, false)
+        .await
+        .map_err(|e| handle_mutiny_err(e))?;
+    Ok(Json(json!("ok")))
+}
+
+#[derive(Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct CreateInvoiceRequest {
+    amount_sat: u64,
+    label: Option<String>,
+}
+
+#[derive(Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct CreateInvoiceResponse {
+    amount_sat: u64,
+    payment_hash: String,
+    serialized: String,
+}
+
+pub async fn create_invoice(
+    Extension(state): Extension<State>,
+    Form(request): Form<CreateInvoiceRequest>,
+) -> Result<Json<Value>, (StatusCode, Json<Value>)> {
+    let label = match request.label {
+        Some(label) => vec![label.to_string()],
+        None => Vec::new(),
+    };
+
+    let invoice = state
+        .mutiny_wallet
+        .create_invoice(request.amount_sat, label)
+        .await
+        .map_err(|e| handle_mutiny_err(e))?;
+
+    let amount_sat = match invoice.amount_sats {
+        Some(amount) => amount,
+        None => {
+            return Err((
+                StatusCode::INTERNAL_SERVER_ERROR,
+                Json(json!({"error": "unable to create invoice"})),
+            ))
+        }
+    };
+
+    let pr = match invoice.bolt11 {
+        Some(amount) => amount,
+        None => {
+            return Err((
+                StatusCode::INTERNAL_SERVER_ERROR,
+                Json(json!({"error": "unable to create invoice"})),
+            ))
+        }
+    }
+    .to_string();
+
+    let invoice = CreateInvoiceResponse {
+        amount_sat,
+        payment_hash: invoice.payment_hash.to_string(),
+        serialized: pr,
+    };
+
+    Ok(Json(json!(invoice)))
+}
+
+#[derive(Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct PayInvoiceRequest {
+    amount_sat: Option<u64>,
+    invoice: Bolt11Invoice,
+}
+
+#[derive(Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct PayInvoiceResponse {
+    recipient_amount_sat: u64,
+    routing_fee_sat: u64,
+    payment_hash: String,
+    payment_preimage: String,
+}
+
+pub async fn pay_invoice(
+    Extension(state): Extension<State>,
+    Form(request): Form<PayInvoiceRequest>,
+) -> Result<Json<Value>, (StatusCode, Json<Value>)> {
+    let invoice = state
+        .mutiny_wallet
+        .pay_invoice(&request.invoice, request.amount_sat, Vec::new())
+        .await
+        .map_err(|e| handle_mutiny_err(e))?;
+
+    let amount_sat = match invoice.amount_sats {
+        Some(amount) => amount,
+        None => {
+            return Err((
+                StatusCode::INTERNAL_SERVER_ERROR,
+                Json(json!({"error": "unable to pay invoice"})),
+            ))
+        }
+    };
+
+    let fees = match invoice.fees_paid {
+        Some(fee) => fee,
+        None => {
+            return Err((
+                StatusCode::INTERNAL_SERVER_ERROR,
+                Json(json!({"error": "unable to pay invoice"})),
+            ))
+        }
+    };
```

Comment:
You do this style a bunch, it is cleaner to do `let fees = invoice.fees_paid.ok_or((StatusCode::INTERNAL_SERVER_ERROR, Json(json!({"error": "unable to pay invoice"})))?`